### PR TITLE
Trying to fix issue 13

### PR DESCRIPTION
**Pull request by karussell**
Now here I stumbled over a problem that the time difference is 0 which leads to an infinite transitionLogProbability. Probably we have to handle those cases separately?

```java
if (timeDiff == 0) {
    return 0;
}
Double transitionMetric = normalizedTransitionMetric(routeLength, linearDistance, timeDiff);
return Distributions.logExponentialDistribution(beta, transitionMetric);
```

Related #13

### CODE REVIEW

**Comment by karussell** [comment]
See the proposed simple fix. But maybe we should throw an error at this point and avoid passing 0 to this method?

File: matching-core/src/main/java/com/graphhopper/matching/util/HmmProbabilities.java [hunk 1]
```diff
@@ -54,34 +52,40 @@ public HmmProbabilities(double sigma, double beta) {
...
+     * consecutive map matching candidates.
+     * @param linearDistance Linear distance [m] between two consecutive GPS
+     * measurements.
+     * @param timeDiff time difference [s] between two consecutive GPS
+     * measurements.
      */
     public double transitionLogProbability(double routeLength, double linearDistance,
             double timeDiff) {
+        if (timeDiff == 0) {
+            return 0;
```
Comment (karussell) [inline matching-core/src/main/java/com/graphhopper/matching/util/HmmProbabilities.java]: Here is the fix

**Comment by michaz** [comment]
These are _logarithms_ of probabilities. Returning 0 means "for sure". Returning minus infinity is less wrong than returning 0, because it means "never", and the probability of being at two different places with time difference 0 is 0.

**Comment by kodonnell** [comment]
> the probability of being at two different places with time difference 0 is 0

so we should check if they are actually at the same place too. If they are - very high transition probability. If they aren't, then low.

Alternatively:

- throw an error. This is what I do in my case, as one should not get such data - and I don't want to be unknowingly processing erroneous data, etc.
- filter out such entries - which probably makes sense to go [here](https://github.com/graphhopper/map-matching/blob/1ea75653b174598aa4711a276f6759c92d9f5363/matching-core/src/main/java/com/graphhopper/matching/MapMatching.java#L247-L267). We can either add this as (configurable) functionality, or allow the user to override the method if they want custom filtering (which gives us/them much more flexibility).

**Comment by michaz** [comment]
Maybe filter them out if they are the same location, and handle them like any other broken chain (possibly with a helpful comment: "you have identical timestamps on different points") otherwise, added by the API/web layer. 

But to be continuous with the normal case, at least the library function should just report it as a broken chain.

**Comment by karussell** [comment]
@michaz would you mind to open a new issue for this? Maybe we should be strict in the core and for the web stuff we just attach artificial timestamps with a speed of ~20km/h (at least per default)?

**Pull request merged**